### PR TITLE
br: add checkpoint for repairing ingest index

### LINKED ISSUE #43893: br checkpoint for log restore is not compatible with repairing ingest index
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. run log restore
2. failed after one ingest index is dropped and before the ingest index is created.
3. retry the br

### 2. What did you expect to see? (Required)
the dropped ingest index should be created in the second execution.
### 3. What did you see instead (Required)
the dropped ingest index is skipped.
### 4. What is your TiDB version? (Required)
v7.1.0

### PR DESCRIPTION
**Pull request by Leavrth**
### What problem does this PR solve?

Issue Number: close #43893 

Problem Summary:
br checkpoint for log restore is not compatible with repairing ingest index
### What is changed and how it works?
add the fixed sqls into the checkpoint mode.
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

the first execution
```
GO_FAILPOINTS="github.com/pingcap/tidb/br/pkg/restore/failed-before-create-ingest-index=1*return(false)->return(true)" ./br restore point -s ...
Restore Meta Files <-----------------------------------------------------------------------------------------> 100.00%
Restore KV Files <-------------------------------------------------------------------------------------------> 100.00%
repair ingest index zips2 for table test.pairs7  ... DONE
repair ingest index i2 for table test.pairs7   ABORTED
[2023/05/17 10:06:54.373 +08:00] [INFO] [collector.go:77] ["restore log failed summary"] [error="failed to repair ingest index: failed before create ingest index"] 
```
the second execution
```
GO_FAILPOINTS="github.com/pingcap/tidb/br/pkg/restore/failed-before-create-ingest-index=1*return(false)->return(true)" ./br restore point -s ...
Restore Meta Files <-----------------------------------------------------------------------------------------> 100.00%
Restore KV Files <-------------------------------------------------------------------------------------------> 100.00%
repair ingest index zips2 for table test.pairs7 ... SKIPPED DUE TO CHECKPOINT MODE
repair ingest index i2 for table test.pairs7  ... DONE
repair ingest index i1 for table test.pairs2   ABORTED
[2023/05/17 10:08:02.975 +08:00] [INFO] [collector.go:77] ["restore log failed summary"] [error="failed to repair ingest index: failed before create ingest index"]
```
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/restore/client.go [hunk 1]
```diff
@@ -3160,34 +3161,46 @@ const (
...
+		if err != nil {
+			return sqls, false, errors.Trace(err)
+		}
+		if exists {
+			checkpointSQLs, err := checkpoint.LoadCheckpointIngestIndexRepairSQLs(ctx, rc.storage, taskName)
+			if err != nil {
+				return sqls, false, errors.Trace(err)
+			}
+			sqls = checkpointSQLs.SQLs
+			log.Info("[ingest] load ingest index repair sqls from checkpoint")
```
Comment (3pointer) [inline br/pkg/restore/client.go:3182]: shall we print SQL in logs?

File: br/pkg/restore/client.go [hunk 2]
```diff
@@ -3209,20 +3222,114 @@ func (rc *Client) RepairIngestIndex(ctx context.Context, ingestRecorder *ingestr
...
+		if fromCheckpoint {
+		NEXTINDEX:
+			for _, idx := range tableInfo.Indices() {
+				indexInfo := idx.Meta()
+				if indexInfo.ID == sql.IndexID {
+					// the original index id is not dropped
+					oldIndexIDFound = true
+					break NEXTINDEX
+				}
+				if indexInfo.Name.O == sql.IndexName {
```
Comment (3pointer) [inline br/pkg/restore/client.go:3281]: I think we need check the index state here

File: br/pkg/restore/client.go [hunk 3]
```diff
@@ -3209,20 +3222,114 @@ func (rc *Client) RepairIngestIndex(ctx context.Context, ingestRecorder *ingestr
...
+	info := dom.InfoSchema()
+
+	sqls, fromCheckpoint, err := rc.generateRepairIngestIndexSQLs(ctx, ingestRecorder, info.AllSchemas(), taskName)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	console := glue.GetConsole(g)
+NEXTSQL:
+	for _, sql := range sqls {
```
Comment (3pointer) [inline br/pkg/restore/client.go:3264]: can we simplify this for loop?
for example. `break NEXTINDEX` can be `break`

**Review by YuJuncen — APPROVED** [review]

**Review by BornChanger — APPROVED** [review]

**Pull request merged**